src/Library/demos/Text Input/main.blp

```diff
@@ -0,0 +1,309 @@
+using Gtk 4.0;
+using Adw 1;
+
+Adw.Clamp {
+  Box {
+    orientation: vertical;
+
+    Label {
+      label: "Text Input";
+      margin-top: 10;
+      margin-bottom: 10;
+
+      styles [
+        "title-1",
+      ]
+    }
+
+    Label {
+      label: "Widgets used for reading text input";
+      margin-bottom: 10;
+    }
+
+    Label {
+      label: "Entry";
+      margin-top: 10;
+      halign: start;
+
+      styles [
+        "heading",
+      ]
+    }
+
+    Label {
+      label: "A simple but versatile widget that allows single line text entry and editing.";
+      margin-top: 10;
+      margin-bottom: 10;
+      halign: start;
+
+      styles [
+        "dim-label",
+      ]
+    }
+
+    FlowBox {
+      homogeneous: true;
+      row-spacing: 20;
+
+      Box {
+        orientation: vertical;
+
+        Label {
+          label: "Regular";
+          margin-bottom: 10;
+        }
+
+        Entry entry {
+
+        }
+      }
+
+      Box {
+        orientation: vertical;
+
+        Label {
+          label: "Overwritable";
+          margin-bottom: 10;
+        }
+
+        Entry {
+          overwrite-mode: true;
+          buffer: bind entry.buffer;
+        }
+      }
+
+      Box {
+        orientation: vertical;
+
+        Label {
+          label: "Progress Bar";
+          margin-bottom: 10;
+
+        }
+
+        Entry progress_entry{
+          progress-fraction: 0;
+          buffer: bind entry.buffer;
+        }
+      }
+
+      Box {
+        orientation: vertical;
+
+        Label {
+          label: "Placeholder Text";
+          margin-bottom: 10;
+        }
+
+        Entry {
+          placeholder-text: "Enter text here";
+          buffer: bind entry.buffer;
```

Comment:
It's a cool effect to synchronize all the buffers, but I'm afraid it will confuse newcomers
In general, for library entries that are about widgets — we should aim for copy pasting to just work.

Reply:
Okay I'll remove it. Also what do you mean by copy pasting to just work?

Reply:
Sorry, it wasn't clear.

I meant that the user should be able to copy the definition for a widget and paste it somewhere else. 
If we have `buffer: bind entry.buffer`, they will get an error saying that there is no object `entry` or something like that.